avro-schema-common/build.gradle.kts

```diff
@@ -1,4 +1,7 @@
 plugins {
     id("io.micronaut.build.internal.avro-module")
 }
-
+dependencies {
+    api(mn.jackson.databind)
```

Comment:
- remove
- in modules that use jackson annotations, depend on `jackson-annotations` in `compileOnly` (i think jackson-annotations is exposed through the bom)
- `@Serdeable` on the classes that need to be serialized
- also need annotationProcessor dependency on `serde-processor`